src/Shared/Microsoft.DotNet.GitHub.Authentication/GitHubAppTokenProvider.cs

```diff
@@ -17,7 +17,7 @@ public GitHubAppTokenProvider(IOptions<GitHubTokenProviderOptions> options = nul
             _options = options;
         }
 
-        public string GetAppToken()
+        public virtual string GetAppToken()
```

Comment:
It might make more sent to extract an interface out of this than make a single method overridable.  This is the only method in the class... if you override it, none of the code in the class means anything anymore...

Reply:
As discussed creating another issue for this and will work on that issue as soon as this PR is merged. 

Issue -> https://github.com/dotnet/core-eng/issues/9582